src/pydvl/influence/influence_calculator.py

```diff
@@ -0,0 +1,554 @@
+
+        def func(
+            z_test_numpy: NDArray,
+            x_numpy: NDArray,
+            y_numpy: NDArray,
+            model: InfluenceFunctionModel,
+        ):
+            ups = model.influences_from_factors(
+                self.numpy_converter.from_numpy(z_test_numpy),
+                self.numpy_converter.from_numpy(x_numpy),
+                self.numpy_converter.from_numpy(y_numpy),
+                influence_type=influence_type,
+            )
+            return self.numpy_converter.to_numpy(ups)
+
+        un_chunked_x_shape = [s[0] for s in x.chunks[1:]]
+        x_chunk_sizes = x.chunks[0]
+        z_test_chunk_sizes = z_test_factors.chunks[0]
+        blocks = []
+        block_shape: Tuple[int, ...]
+
+        for z_test_chunk, z_test_chunk_size in zip(
+            z_test_factors.to_delayed(), z_test_chunk_sizes
+        ):
+            row = []
+            for x_chunk, y_chunk, chunk_size in zip(
+                x.to_delayed(), y.to_delayed(), x_chunk_sizes
+            ):
+                if influence_type == InfluenceType.Perturbation:
+                    block_shape = (z_test_chunk_size, chunk_size, *un_chunked_x_shape)
+                elif influence_type == InfluenceType.Up:
+                    block_shape = (z_test_chunk_size, chunk_size)
+                else:
+                    raise UnSupportedInfluenceTypeException(influence_type)
+
+                block_array = da.from_delayed(
+                    delayed(func)(
+                        z_test_chunk.squeeze().tolist(),
+                        x_chunk.squeeze().tolist(),
+                        y_chunk.squeeze().tolist(),
+                        self.influence_function_model,
+                    ),
+                    shape=block_shape,
+                    dtype=z_test_factors.dtype,
+                )
+
+                if influence_type == InfluenceType.Perturbation:
+                    num_dims = block_array.ndim
+                    new_order = tuple(range(2, num_dims)) + (0, 1)
+                    block_array = block_array.transpose(*new_order)
+
+                row.append(block_array)
+            blocks.append(row)
+
+        values_array = da.block(blocks)
+
+        if influence_type == InfluenceType.Perturbation:
+            num_dims = values_array.ndim
+            new_order = (num_dims - 2, num_dims - 1) + tuple(range(num_dims - 2))
+            values_array = values_array.transpose(*new_order)
+
+        return values_array
+
+    @staticmethod
+    def _get_client() -> Optional[distributed.Client]:
+        try:
+            return distributed.get_client()
+        except ValueError:
+            return None
+
+
+class BlockAggregator(Generic[TensorType], ABC):
+    @abstractmethod
+    def aggregate_nested(
+        self, tensors: Generator[Generator[TensorType, None, None], None, None]
+    ):
+        """Overwrite this method to aggregate provided blocks into a single tensor"""
```

Comment:
The docstring doesn't help me understand what the input is like (nested?), and what I should do with it. But I think this function is just a "flatten", right? Then it should be called flatten